worlds/dkc3/Client.py

```diff
@@ -77,9 +77,10 @@ async def dkc3_game_watcher(ctx: Context):
                     # DKC3_TODO: Handle non-included checks
                     new_checks.append(loc_id)
 
-        save_file_name = await snes_read(ctx, DKC3_FILE_NAME_ADDR, 0x5)
-        if save_file_name is None or save_file_name[0] == 0x00:
-            # We have somehow exited the save file
+        verify_save_file_name = await snes_read(ctx, DKC3_FILE_NAME_ADDR, 0x5)
+        if verify_save_file_name is None or verify_save_file_name[0] == 0x00 or verify_save_file_name != save_file_name:
+            ctx.rom = None
```

Comment:
I think trying to validate the rom before setting ctx.rom = None would be better to avoid reconnecting to the room if the rom didn't change

Reply:
The goal in this scenario is to raise red flags and get out of our current state if we detect a change. The save file name changing means:
1) The game crashed
2) A reset was triggered in the middle of this function, or
3) a savestate was loaded in the middle of this function.
What would be the appropriate behavior in this case, if not to try to start the process over from the beginning?